### PR TITLE
Use GraphCanvas node palette in EMFX context menu

### LINKED ISSUE #10963: Remove "Anim Graph Palette" window and move functionality (No dependency)
[Migrated from JIRA LYN-14133]

Please see the attached video called "CurrentState_HowToCreateNodes.mp4" to see the currently available ways on how to create nodes in the anim graph.

The aim of this task is to get rid of the "Anim Graph Palette" (first shown in the video) and improve the context menu based version (shown afterwards in the video).

The new workshow shall be like in the "Script Canvas" tool. Please see the video "NewWay_LikeInScriptCanvas.mp4" on how that shall work.

**Acceptance Criteria:**

- EMStudio::NodePaletteWidget code is removed from the project including the NodePaletteWidget.h/.cpp.
- The context menu provides a search widget to search for the available nodes as well as a tree view to easily select them (like in the previous node palette widget).

**Additional Info:**

https://user-images.githubusercontent.com/82228511/181770313-2a7586d2-154b-4e8d-a3de-eb62cf02ded1.mp4

https://user-images.githubusercontent.com/82228511/181770324-bc63c468-e6b3-489e-a795-f6e58b8bb124.mp4

### PR DESCRIPTION
**Pull request by MiKom**
- Original node palette dock window is still there but hidden by default
  in the layout
- Decision of what font color should be used for inactive nodes in the
  NodePaletteTreeWidget was moved from model to the delegate. The model
  doesn't have the information about the default text color so it
  wouldn't what color to make transparent. I investigated delegating
  the decision of the inactive color to Qt through passing the
  inactivity info from model's ::flags() method (Qt::ItemIsEnabled).
  However, this caused some ugly artifacts. Inactive tree items where
  highlighted partially (on the expand triangle)
- Nodes don't yet have the colored rectangle next to them. For this
  I'd like to use icon engine developed for #10961. Thus it will be done
  in a follow-up PR.

Fixes #10963

Signed-off-by: Miłosz Kosobucki <milosz@kosobucki.pl>

## What does this PR do?

Adds the node palette tree item to context menu in EMotionFX. Same type as in {Script,Landscape,Material}Canvas

## How was this PR tested?

Unit tests were updated to use the new API and I tested it manually.

### CODE REVIEW

**Comment by MiKom** [comment]
Please let me know if you want me to add the icons in the tree in this PR or can we get this one through and add icons later.

**Comment by MiKom** [comment]
That's how it looks:

https://user-images.githubusercontent.com/104767/186953495-41d966ed-22f0-4b72-9694-7a3d1d084227.mp4

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/AnimGraphPlugin.h [hunk 1]
```diff
@@ -63,6 +65,8 @@ namespace EMStudio
     class TimeViewPlugin;
     class NavigationHistory;
 
+    static constexpr GraphCanvas::EditorId AnimGraphEditorId = AZ_CRC("AnimGraphEditor",  0xf44fce98);
```
Comment (cgalvan) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/AnimGraphPlugin.h:68]: Nit: Let the compiler compute the hash for us

```suggestion
    static constexpr GraphCanvas::EditorId AnimGraphEditorId = AZ_CRC_CE("AnimGraphEditor");
```

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/BlendGraphWidget.cpp [hunk 2]
```diff
@@ -34,20 +34,66 @@
...
+#include <GraphCanvas/Widgets/GraphCanvasTreeItem.h>
+#include <GraphCanvas/Widgets/NodePalette/TreeItems/NodePaletteTreeItem.h>
 // qt includes
 #include <QDropEvent>
 #include <QMessageBox>
 #include <QMimeData>
 #include <QMouseEvent>
 #include <QToolTip>
 #include <QWidget>
+#include <QDebug>
```
Comment (cgalvan) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/BlendGraphWidget.cpp:46]: Nit: Can remove `QDebug`

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ContextMenu.cpp [hunk 3]
```diff
@@ -22,50 +22,46 @@
...
 
 
 namespace EMStudio
 {
-    // Fill the given menu with anim graph objects that can be created inside the current given object and category.
-    void BlendGraphWidget::AddAnimGraphObjectCategoryMenu(AnimGraphPlugin* plugin, QMenu* parentMenu,
-        EMotionFX::AnimGraphObject::ECategory category, EMotionFX::AnimGraphObject* focusedGraphObject)
+    // Fill the node palette tree view with the anim graph objects that can be created inside the current given object and category.
+    void BlendGraphWidget::AddCategoryToNodePalette(EMotionFX::AnimGraphNode::ECategory category, GraphCanvas::NodePaletteTreeItem* rootNode,
+                               EMotionFX::AnimGraphObject* focusedGraphObject)
```
Comment (cgalvan) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ContextMenu.cpp:35]: Nit: Looks like some extra whitespace here?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/AnimGraphPlugin.h (hunk 1 above)
Comment (cgalvan) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/AnimGraphPlugin.h:68]: Actually the hash was a mismatch which failed the validation on AR, so even more of a reason to use `AZ_CRC_CE` instead :)

```
VALIDATOR_FAILED: CrcValidator /data/workspace/o3de/Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/AnimGraphPlugin.h::CrcValidator FAILED - Source file contains a CRC mismatch!
[2022-08-29T18:10:46.857Z]          AZ_CRC("AnimGraphEditor", 0xf44fce98), expected value 0xfdba7ae7
```

**Review by cgalvan — CHANGES_REQUESTED** [review]
The above comments are mostly nits except the CRC one needs to be updated to pass validation

**Review by cgalvan — APPROVED** [review]

**Comment by cgalvan** [comment]
> Hey @cgalvan Thanks for review and all the comments. I adressed them in the new commits.

Thanks @MiKom ! I've approved and kicked off a new AR.

**Review by lsemp3d — APPROVED** [review]

**Review by alexmontAmazon — APPROVED** [review]

**Pull request merged**